app/src/main/java/com/github/swent/echo/ui/navigation/Navigation.kt

```diff
@@ -0,0 +1,60 @@
+package com.github.swent.echo.ui.navigation
+
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+
+/** The routes to the screens of the app */
+enum class Routes {
+    SIGN_IN,
+    MAP,
+}
+
+/**
+ * This class reprensent the different navigation actions
+ *
+ * @param navController the navigation controller
+ */
+class navigationActions(val navController: NavHostController) {
+
+    // navigate to one of the routes available in Routes
+    fun navigateTo(route: Routes) {
+        navController.navigate(route.name)
+    }
+
+    // navigate to the previous screen
+    fun goBack() {
+        navController.navigateUp()
+    }
+}
+
+/**
+ * Navigation composable, it display the relevant screen based on the route and pass an instance of
+ * navigationActions to the screens
+ *
+ * @param navController the navigation controller, created if none is given
+ */
+@Composable
+fun appNavigationHost(
```

Comment:
Nitpicking: The convention is to start composables with a capital letter: `AppNavigationHost`.
Nitpicking: I think it could make sense to extract this into its own file.

Reply:
done